### ISSUE TITLE
ttn decoded data is different to display/serial console data

### ISSUE THREAD

**Issue by muckisg** [kizniche/ttgo-tbeam-ttn-tracker#37]
Hi,

i was asked to start a new issue. I am working with a T-Beam V1.0 and realized all steps from the readme file.
My T-Beam display and serial console show the right gps coordinates but using the decoder script from the readme with ttn show different results.

**TTN output:**

![image](https://user-images.githubusercontent.com/617491/112335122-41c37180-8cbc-11eb-94b7-69fb9a2b041f.png)

**Serial console output:**

![image](https://user-images.githubusercontent.com/617491/112335319-6b7c9880-8cbc-11eb-99b1-2aab7b1ccb7b.png)

any ideas?

Thanks in advance,

Maurice aka muckisg

**Comment by muckisg** [comment]
Yes, sure i do as far as i can. It is strange to me that the results in ttn differ from packet to packet, so that the payload seems to be interpreted completly wrong but i guess i would not be the first person to find this problem then. Could it be a byte order problem?
Thanks for your quick response by the way! :-)

**Comment by fdhosting** [comment]
I had the same problem, solved it by using another decoder, check your source and look if there more versions of the decoder inside 😀

Regards,

Arthur

Verstuurd vanaf mijn iPhone

**Comment by muckisg** [comment]
Would you share your decoder? If this is the problem it is a structural one so that every user should have the problem as far as i can  tell. Thanks in advance, MuckiSG

**Comment by fdhosting** [comment]
Please check this:

https://github.com/cyberman54/ESP32-Paxcounter

Really works nice, all in one package 😀

**Comment by muckisg** [comment]
FYI:
Payload within ttn_send: C8 BD 3E 85 09 0B 00 A7 12 06  <-correct
Payload in TTN console: DB DB 7D 33 F7 F0 68 50 F0 84  <- bad

Because of the fact the payload is correct within the ttn_send function, the error has to be somewhere within the trasmission or ttn handling. The payload ttn receives is definitely wrong! When i send the payload "00 01 02 03 04 05 06 07 08 09" i get totaly different resunlts on the ttn end an even changing results.

So, your software seams to be correct but do you have an idea where the error is?

Tried a second board aswell with the same results, so no hardware error.

thx, Maurice
